source/GameGraphRoot.cpp

```diff
@@ -191,6 +194,37 @@ void GameGraphRoot::update(float timestep) {
 			breachModel->getSprite()->setPosition(breachPos);
 		}
 	}
+	for (int i = 0; i < doors.size(); i++) {
+		std::shared_ptr<DoorModel> doorModel = doors.at(i);
+		if (doorModel->getSprite() == nullptr) {
```

Comment:
This should be done in `init()`. Once a doorModel is instantiated it will never have a null sprite. Also let's not even mention the issue of model owning reference to node.

Reply:
I changed these things in recent commits. Let me know what you think.